realm/realm-library/src/androidTest/kotlin/io/realm/RealmModelManagedSetTester.kt

```diff
@@ -0,0 +1,397 @@
+            managedInitializedSet = realm.copyToRealmOrUpdate(unmanagedInitializedSet)
+            managedNotPresentValue = realm.copyToRealmOrUpdate(unmanagedNotPresentValue)
+        }
+
+        this.managedTester = ManagedSetTester(
+                testerName = testerName,
+                mixedType = mixedType,
+                setGetter = setGetter,
+                setSetter = setSetter,
+                managedSetGetter = managedSetGetter,
+                managedCollectionGetter = managedCollectionGetter,
+                initializedSet = managedInitializedSet,
+                notPresentValue = managedNotPresentValue,
+                toArrayManaged = toArrayManaged
+        )
+
+        this.managedTester.setUp(config, looperThread)
+    }
+
+    override fun tearDown() {
+        managedTester.tearDown()
+        this.realm.close()
+    }
+
+    override fun isManaged() = managedTester.isManaged()
+
+    override fun isValid() = managedTester.isValid()
+
+    override fun isFrozen() = managedTester.isFrozen()
+
+    override fun size() = managedTester.size()
+
+    override fun isEmpty() = managedTester.isEmpty()
+
+    override fun contains() {
+        managedTester.contains()
+
+        // Test with unmanaged realm objects
+        val set = initAndAssertEmptySet()
+        realm.executeTransaction {
+            // Check throws exception when unmanaged values are passed
+            assertFailsWith<IllegalArgumentException>("Unmanaged objects not permitted") {
+                set.contains(unmanagedNotPresentValue)
+            }
+
+            assertFalse(set.contains(null))
+        }
+
+        // Test with objects from another realm
+        looperThread.runBlocking {
+            Realm.getInstance(realm.configuration).use { alternativeRealm ->
+                alternativeRealm.executeTransaction {
+                    val value = alternativeRealm.copyToRealmOrUpdate(unmanagedNotPresentValue)
+
+                    assertFailsWith<IllegalArgumentException>("Cannot pass an object from another Realm instance") {
+                        set.contains(value)
+                    }
+                }
+            }
+            looperThread.testComplete()
+        }
+    }
+
+    override fun iterator() = managedTester.iterator()
+
+    override fun toArray() = managedTester.toArray()
+
+    override fun toArrayWithParameter() = managedTester.toArrayWithParameter()
+
+    override fun add() {
+        // Test with managed objects
+        managedTester.add()
+
+        // Test with unmanaged objects
+        val set = initAndAssertEmptySet()
+        realm.executeTransaction {
+            // Adding a value for the first time returns true
+            unmanagedInitializedSet.forEach { value ->
+                assertTrue(set.add(value))
+            }
+            // Adding an existing value returns false
+            unmanagedInitializedSet.forEach { value ->
+                assertFalse(set.add(value))
+            }
```

Comment:
Is it caught in the preprocessor that we do not support embedded objects?

Reply:
It would throw in runtime, when validating the schema. I have added a check in the processor to catch it.